application-ckeditor-plugins/src/main/resources/xwiki-macro/macroEditor.js

```diff
@@ -532,14 +532,19 @@ define('macroParameterTreeDisplayer', ['jquery', 'l10n!macroEditor'], function($
         if (matchedOption.length > 0) {
           // Use the canonical value.
           value = matchedOption.val();
+        } else if ('select-multiple' === firstInputType) {
+          // Add the missing options if the select allows multiple values.
+          value.split(',').forEach(function (entry) {
+            $('<option></option>').val(entry).text(entry).appendTo(valueInputs);
+          });
```

Comment:
I would do it differently:

* after checking above if the input is a select set ``value`` to ``value.split(',')``
* then iterate over each value and do what was done before for a single value, i.e. check if the option exists and if not then add it; you'll have to refactor ``matchesParameterValue`` to get the value as input, and you also need to update the value in the value array if the option exists (needed when the match was case insensitive)
* at the end, see below, check if the value is an array